src/Interpreters/executeQuery.cpp

```diff
@@ -378,6 +386,39 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
             /// TODO: parser should fail early when max_query_size limit is reached.
             ast = parseQuery(parser, begin, end, "", max_query_size, settings.max_parser_depth);
         }
+        #if USE_PRQL
+        else if (settings.dialect == Dialect::prql && !internal)
+        {
+            char * prql_query = new char[max_query_size + 1];
+            if (prql_query == nullptr)
+            {
+                throw Exception(ErrorCodes::CANNOT_ALLOCATE_MEMORY, "Not enough RAM");
+            }
+            memset(prql_query, 0, max_query_size + 1);
+            memcpy(prql_query, begin, end - begin);
+            char * sql_query = new char[max_query_size + 1];
+            if (sql_query == nullptr)
+            {
+                throw Exception(ErrorCodes::CANNOT_ALLOCATE_MEMORY, "Not enough RAM");
+            }
+            memset(sql_query, 0, max_query_size + 1);
+            int res = to_sql(prql_query, sql_query);
```

Comment:
This is major - **there is no boundary checks for the output buffer**.

Reply:
Refactored, see my other comment.